internal/workerutil/observability.go

```diff
@@ -75,3 +81,58 @@ func newOperations(observationContext *observation.Context, prefix string, keys,
 		handle: op("Handle"),
 	}
 }
+
+// lenientConcurrencyGaugeInterval is the interval at which a lenient
+// concurrency gauge's value is updated with the current window's maximum
+// value.
+const lenientConcurrencyGaugeInterval = time.Second * 5
+
+// newLenientConcurrencyGauge creates a new gauge-like object that emits
+// (into the given gauge) the maximum value over the last five seconds.
+// This method begins an immortal background routine.
+//
+// This gauge should be used to smooth-over the randomness sampled by
+// Prometheus by emitting the aggregate we'll likely be using with this
+// type of data directly.
+//
+// Without wrapping the numJobs gauge in this object, we tend to sample
+// zero handlers consistently when jobs are short-lived (< 500ms).
+// Keeping the max in memory gives us the value that we actually want
+// over time, and indicates an accurate level of concurrency.
+func newLenientConcurrencyGauge(gauge prometheus.Gauge) Gauge {
+	ch := make(chan float64)
+	go runLenientConcurrencyGauge(gauge, ch)
+
+	return &channelGauge{ch: ch}
+}
+
+func runLenientConcurrencyGauge(gauge prometheus.Gauge, ch <-chan float64) {
+	max := float64(0)
+	value := float64(0)
+	ticker := time.NewTicker(lenientConcurrencyGaugeInterval)
+
+	for {
+		select {
+		case <-ticker.C:
+			gauge.Set(max)
+			max = 0
+
+		case update, ok := <-ch:
+			if !ok {
+				return
+			}
+
+			value += update
+			if value > max {
```

Comment:
Its a slightly different semantic, but instead this could be like this to capture negative values?
```suggestion
			if Abs(value) > max {
```

Reply:
Updated docs to specify this is to track concurrency only and negative values will produce undefined behavior.